sensor/common/networkflow/manager/purger.go

```diff
@@ -0,0 +1,100 @@
+package manager
+
+import (
+	"time"
+
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/timestamp"
+	flowMetrics "github.com/stackrox/rox/sensor/common/networkflow/metrics"
+)
+
+func (m *networkFlowManager) purgeStaleEndpoints(tickerC <-chan time.Time) {
+	for {
+		select {
+		case <-m.stopper.Flow().StopRequested():
+			return
+		case <-tickerC:
+			maxAge := env.EnrichmentPurgerTickerMaxAge.DurationSetting()
+			start := time.Now()
+			concurrency.WithLock(&m.activeEndpointsMutex, func() {
+				log.Debug("Purging active endpoints")
+				purgeActiveEndpointsNoLock(maxAge, m.activeEndpoints, m.clusterEntities)
+			})
```

Comment:
Why do we need a special case for `activeEndpoints`? Is it purging the `hostConnections.endpoint` not enough? My understanding is that they should behave like `activeConnections` and be removed in the next tick if we clean `hostConnections`.

Reply:
I will check whether `activeConnections` are cleaned. I thought that once you add something to `activeConnections` (and `activeEndpoints`) it will stay there until we delete that manually - and for the unclosed endpoints we would never delete it. 
I will double check that if things can stuck in `activeEndpoints` and adapt the code if needed.

Reply:
Unfortunately, disabling the purger for activeEndpoints makes them accumulate in memory over time. We also must purge them.
![Screenshot 2025-04-08 at 15 57 40](https://github.com/user-attachments/assets/db86ccfc-13da-4168-b275-232fd9041b1f)

Reply:
I readded the purger. 
I will handle the issue of things being copied from hostConnections to activeConnections in a follow-up.